controlplane/internal/controller/agentcontrolplane_controller.go

```diff
@@ -274,7 +274,8 @@ func (r *AgentControlPlaneReconciler) ensureClusterDeployment(
 ) error {
 	if acp.Status.ClusterDeploymentRef == nil {
 		clusterDeployment := assistedinstaller.GetClusterDeploymentFromConfig(acp, clusterName)
-		if err := r.Create(ctx, clusterDeployment); err != nil && !apierrors.IsAlreadyExists(err) {
+		controllerutil.SetOwnerReference(acp, clusterDeployment, r.Scheme)
```

Comment:
Linter will not be happy about the unhandled return value

Reply:
Gotcha, thanks!